code/modules/mob/living/simple_animal/hostile/terror_spiders/actions.dm

```diff
@@ -19,17 +19,84 @@
 	user.FindWrapTarget()
 	user.DoWrap()
 
+// ---------- GREEN ACTIONS
+
+/datum/action/innate/terrorspider/greeneggs
+	name = "Lay Green Eggs"
+	icon_icon = 'icons/effects/effects.dmi'
+	button_icon_state = "eggs"
+
+/datum/action/innate/terrorspider/greeneggs/Activate()
+	var/mob/living/simple_animal/hostile/poison/terror_spider/green/user = owner
+	user.DoLayGreenEggs()
+
+// ---------- PRINCE ACTIONS
+
+/datum/action/innate/terrorspider/princesmash
+	name = "Smash Welded Vent"
+	icon_icon = 'icons/atmos/vent_pump.dmi'
+	button_icon_state = "map_vent"
+
+/datum/action/innate/terrorspider/princesmash/Activate()
+	var/mob/living/simple_animal/hostile/poison/terror_spider/prince/user = owner
+	user.DoPrinceSmash()
+
+
+// ---------- QUEEN ACTIONS
+
+/datum/action/innate/terrorspider/queen/queennest
+	name = "Nest"
+	icon_icon = 'icons/mob/terrorspider.dmi'
+	button_icon_state = "terror_queen"
+
+/datum/action/innate/terrorspider/queen/queennest/Activate()
+	var/mob/living/simple_animal/hostile/poison/terror_spider/queen/user = owner
+	user.NestMode()
+
+/datum/action/innate/terrorspider/queen/queensense
+	name = "Hive Sense"
+	icon_icon = 'icons/mob/actions.dmi'
+	button_icon_state = "mindswap"
+
+/datum/action/innate/terrorspider/queen/queensense/Activate()
+	var/mob/living/simple_animal/hostile/poison/terror_spider/queen/user = owner
+	user.DoHiveSense()
+
+/datum/action/innate/terrorspider/queen/queeneggs
+	name = "Lay Queen Eggs"
+	icon_icon = 'icons/effects/effects.dmi'
+	button_icon_state = "eggs"
+
+/datum/action/innate/terrorspider/queen/queeneggs/Activate()
+	var/mob/living/simple_animal/hostile/poison/terror_spider/queen/user = owner
+	user.LayQueenEggs()
+
+/datum/action/innate/terrorspider/queen/queenfakelings
+	name = "Fake Spiderlings"
+	icon_icon = 'icons/effects/effects.dmi'
+	button_icon_state = "spiderling"
+
+/datum/action/innate/terrorspider/queen/queenfakelings/Activate()
+	var/mob/living/simple_animal/hostile/poison/terror_spider/queen/user = owner
+	user.QueenFakeLings()
 
 // ---------- WEB
 
 /mob/living/simple_animal/hostile/poison/terror_spider/proc/Web()
+	var/turf/mylocation = loc
 	visible_message("<span class='notice'>[src] begins to secrete a sticky substance.</span>")
 	if(do_after(src, 40, target = loc))
-		var/obj/effect/spider/terrorweb/T = locate() in get_turf(src)
-		if(T)
-			to_chat(src, "<span class='danger'>There is already a web here.</span>")
+		if(loc != mylocation)
+			return
```

Comment:
IIRC, there were issues with it being possible to, for example, spin a web while drifting through space. do_after did not seem to prevent this. Hence this check.